lib/logstash/inputs/redis.rb

```diff
@@ -86,6 +103,9 @@ def register
     if @data_type == 'list' || @data_type == 'dummy'
       @run_method = method(:list_runner)
       @stop_method = method(:list_stop)
+    elsif @data_type == 'pattern_list'
```

Comment:
instead of adding another `elsif, could change this to
```
case @data_type
when 'list' || 'dummy'
  ...
when 'pattern_list'
  ..
end
```
to make it a bit more elegant.

Reply:
I'd keep refactorings out of this PR as anything that touches the upstream code will make this rot quicker.

Reply:
Might be true, although it's couple years since the upstream has touched the code the last time.